src/modules/peek/Peek.FilePreviewer/FilePreview.xaml.cs

```diff
@@ -128,5 +135,33 @@ private void PreviewBrowser_NavigationCompleted(WebView2 sender, Microsoft.Web.W
                 BrowserPreviewer.State = PreviewState.Loaded;
             }
         }
+
+        private async Task UpdateImageTooltipAsync()
+        {
+            if (File == null)
+            {
+                return;
+            }
+
+            imageTooltip = string.Empty;
+
+            // Fetch and format available file properties
+            imageTooltip += ReadableStringHelper.FormatResourceString("PreviewTooltip_FileName", File.FileName);
+
+            string fileType = await PropertyHelper.GetFileType(File.Path);
+            imageTooltip += string.IsNullOrEmpty(fileType) ? string.Empty : "\n" + ReadableStringHelper.FormatResourceString("PreviewTooltip_FileType", fileType);
+
+            string dateModified = File.DateModified.ToString();
+            imageTooltip += string.IsNullOrEmpty(dateModified) ? string.Empty : "\n" + ReadableStringHelper.FormatResourceString("PreviewTooltip_DateModified", dateModified);
+
+            Size dimensions = await PropertyHelper.GetImageSize(File.Path);
+            imageTooltip += dimensions.IsEmpty ? string.Empty : "\n" + ReadableStringHelper.FormatResourceString("PreviewTooltip_Dimensions", dimensions.Width, dimensions.Height);
+
+            ulong bytes = await PropertyHelper.GetFileSizeInBytes(File.Path);
+            string fileSize = ReadableStringHelper.BytesToReadableString(bytes);
+            imageTooltip += string.IsNullOrEmpty(fileSize) ? string.Empty : "\n" + ReadableStringHelper.FormatResourceString("PreviewTooltip_FileSize", fileSize);
+
+            OnPropertyChanged(nameof(ImageInfoTooltip));
```

Comment:
If you use an [ObservableProperty] for ImageTooltip, it would work without having to request manually OnPropertyChanged.